app/src/main/java/com/dscout/membranevideoroomdemo/viewmodels/RoomViewModel.kt

```diff
@@ -147,6 +147,19 @@ class RoomViewModel(
             isMicrophoneOn.value = enabled
             room.value?.updateTrackMetadata(it.id(), mapOf("active" to enabled, "type" to "audio"))
         }
+        val p = mutableParticipants[localPeerId]
+        if (p != null) {
+            mutableParticipants[localPeerId] = p.copy(
+                tracksMetadata = p.tracksMetadata + (
```

Comment:
can we have this as a function that returns a new participant? see `Peer.withTrack` in the library. That way we'll reduce a loot of repetition.

Reply:
Done.